### PR TITLE
Support back to 3.6.9

### PR DESCRIPTION
**Pull request by norton120**
## Pull request checklist

Before submitting your PR, please review the following checklist:

- [x] Consider adding a unit test if your PR resolves an issue.
- [x] All new and existing tests pass locally (`palm test`)
- [x] Lint (`palm lint`) has passed locally and any fixes were made for failures
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Breaking changes

- [ ] Check if this pull request introduces a breaking change

## What does this implement/fix? Explain your changes.
Changes to the subprocess interface are not breaking going _forward_ (3.6 => 3.7), but break going backwards from our 
primary 3.8 dev environment. This PR attacks 2 things:
- fixes core compatibility back to python 3.6.9 (the stock python3 for Ubuntu 18.04, so a really common semi-recent version).
- creates an abstracted `run_on_the_metal()` interface so we can protect downstream commands from the same issues   

## Does this close any currently open issues?
it makes a big dent in issue #20 

## Any other comments?
To keep this support clean in testing we should add a make option to build and test in virtualenvs for each minor release we want to cover in CI. it would just need to step through each one, run the tests etc. 

## Where has this been tested?
Locally with python3.6.9, python3.7.0, python3.9.9

**Operating System:**
Ubuntu 18.04

### CODE REVIEW

**Comment by norton120** [comment]
This is only a WIP because I need to add docs for the new context method and lint. otherwise it's gtg

File: palm/utils.py [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+         - errors in the shell are NOT bubbled up by default
+        Args:
+            cmd: the command to run.
+            bubble_error: if True raise a `CalledProcessError<https://docs.python.org/3.6/library/subprocess.html#subprocess.CalledProcessError>`_
+        Returns:
+            Tuple: status code, stdout, stderr
+    """
+    major, minor, patch, _, _ = version_info
+    kwargs = dict(shell=True, check=bubble_error)
+    if major < 3 or minor < 5:
```
Comment (norton120) [inline palm/utils.py:33]: eventually this will probably be more abstracted, this is the only place we care at the moment.

**Comment by norton120** [comment]
No longer a WIP, let's roll this puppy into develop!

**Comment by mrogers-palmetto** [comment]
I definitely understand the value in having this function, however the name of it to me is confusing. "On the metal" is usually used to refer to instruction sets that run directly by the hardware, but in this case, Python is technically nowhere near the metal...
Can we rename it to something that more clearly distinguishes its use from `run_in_docker`, like `run_local`? This, to me, seems like one of the things most users would have difficulty understanding--when to use this function instead of `run_in_docker` for their commands. We could also just call it `run_subprocess` so at least folks know it is meant to be used instead of subprocess and wraps it with some handling.

**Comment by norton120** [comment]
> I definitely understand the value in having this function, however the name of it to me is confusing. "On the metal" is usually used to refer to instruction sets that run directly by the hardware, but in this case, Python is technically nowhere near the metal... Can we rename it to something that more clearly distinguishes its use from `run_in_docker`, like `run_local`? This, to me, seems like one of the things most users would have difficulty understanding--when to use this function instead of `run_in_docker` for their commands. We could also just call it `run_subprocess` so at least folks know it is meant to be used instead of subprocess and wraps it with some handling.

interesting - I've heard people use "on the metal" in the context of containers to refer to running a thing on the host. actually, how about `run_on_host()` ? that's pretty descriptive?

**Comment by jakeberesford-palmetto** [comment]
> > I definitely understand the value in having this function, however the name of it to me is confusing. "On the metal" is usually used to refer to instruction sets that run directly by the hardware, but in this case, Python is technically nowhere near the metal... Can we rename it to something that more clearly distinguishes its use from `run_in_docker`, like `run_local`? This, to me, seems like one of the things most users would have difficulty understanding--when to use this function instead of `run_in_docker` for their commands. We could also just call it `run_subprocess` so at least folks know it is meant to be used instead of subprocess and wraps it with some handling.
> 
> interesting - I've heard people use "on the metal" in the context of containers to refer to running a thing on the host. actually, how about `run_on_host()` ? that's pretty descriptive?

I prefer run_on_host - short and clear!

**Comment by jakeberesford-palmetto** [comment]
This looks good, thanks for forking. I'd like to test on Monday before merging

**Comment by norton120** [comment]
> This looks good, thanks for forking. I'd like to test on Monday before merging

sweet. I'm using this branch now, will keep poking at it all weekend :)

File: palm/utils.py [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+
 def is_cmd_file(filename: str) -> bool:
     return filename.endswith(".py") and filename.startswith("cmd")
 
 
 def cmd_name_from_file(filename: str) -> str:
     return filename[4:-3]
+
+
+def run_on_host(
```
Comment (jakeberesford-palmetto) [inline palm/utils.py:18]: Nit: It'd be nice to move this out of the utils.py "misc drawer" and into a better named module

**Review by jakeberesford-palmetto — APPROVED** [review]
Tested this against docker 3.6, test suite passes, this looks good-to-go!

File: setup.py [hunk 3]
```diff
@@ -21,8 +21,9 @@
 long_description = Path(this_directory, 'README.md').read_text()
 
 setup(
+    python_requires='>=3.6.9',
     name='palmcli',
-    version='2.0.1',  # Don't forget to bump the version in docs/source/conf.py too!
+    version='2.0.2',  # Don't forget to bump the version in docs/source/conf.py too!
```
Comment (jakeberesford-palmetto) [inline setup.py:26]: Let's revert this change, version bumping to be handled in release branches from develop -> main

**Review by mrogers-palmetto — APPROVED** [review]
I like!
